core/src/main/scala/kafka/controller/ControllerEventManager.scala

```diff
@@ -139,4 +140,19 @@ class ControllerEventManager(controllerId: Int,
     }
   }
 
+  private def pollFromEventQueue(): QueuedEvent = {
```

Comment:
This approach seems fine for this particular usage. There are other usage of histogram that all have the same issue. A more complete fix would be to fix the issue at the Histogram level. Could we try patching the Yammer code upstream or at least file an issue? I am thinking that we could track the last time a histogram is updated, if it passes a certain amount of time, we reset on next get() or update().

Reply:
@junrao Two of the biggest issues is that the histogram does not reset when the broker shuts down or when the controller moves. There will not be a next `get()/update()` in these scenarios - should we reset the histogram outside of `ControllerEventThread` instead?

Reply:
What I was thinking is the following. We change Histogram so that it tracks the last time it's updated. If the expiration time has been reached, the histogram will reset on next update. If there is no next update, but the histogram value is being read, we will also reset the value on read.

Reply:
unfortunately, the metrics version kafka uses (v2.2.0) is no longer supported by yammer https://github.com/dropwizard/metrics/issues/1618